code/modules/psionics/complexus/complexus_latency.dm

```diff
@@ -1,4 +1,5 @@
 /datum/psi_complexus/proc/check_psionic_trigger(var/trigger_strength = 0, var/source, var/redactive = FALSE)
+	usr = owner
```

Comment:
`usr` shouldn't be relied upon, even less assigned to, what is the point of this assignment?

Reply:
I thought I saw it used elsewhere in the code, though I suppose that's good reasoning, with this codebase.

The point, from the PR Body:

> Once that was fixed, it caused many RTEs when trying to draw the HUD/screen for the powers. Manually setting the usr in check_psionic_trigger fixes that.

The RTEs caused the screen to be drawn at one-third size, halfway down the right side of the screen. The RTEs were caused by null usr in several screen drawing procs that I didn't note and don't remember now. I didn't want to touch those procs, so I did this.

Reply:
@FluffyGhoster Do you have any alternate suggestions? The usr is dropped some time in the callback, not sure how to reset the reference without setting it here.

Reply:
You should follow the calls and see where it happens, and that would also tell you why it happens

Reply:
I've followed the calls, or else I wouldn't have known where to set `usr` to fix the issue. It's dropped because of the callback. Here is the call stack, break point where `usr = owner`:
![image](https://github.com/Aurorastation/Aurora.3/assets/89751433/d7cca39c-77cd-4f54-93cc-528647673cf1)

Here is the call stack when `usr` isn't set in `check_psionic_trigger` and it fails:

![image](https://github.com/Aurorastation/Aurora.3/assets/89751433/9eaef776-f6e5-48a1-aed4-329d52ad9baa)


Below is where the callback is set. From what I can see, anything that hits `InvokeAsync` doesn't have a `usr` value. 
Is there a way to modify the callback so that it does?

https://github.com/Aurorastation/Aurora.3/blob/63b56177e6ab2b9cb6753aac94943c8d2992b510/code/modules/mob/living/carbon/human/human_helpers.dm#L208

Amusingly, I've found an assignment that seems related:

https://github.com/Aurorastation/Aurora.3/blob/56a18f094aea377eb0004624783af61c3bb0aa90/code/modules/spell_system/spells/spells.dm#L10

I'm asking for alternate solutions, since there is a problem with the one I've come up with, and I do not know how else to solve it.

Reply:
I don't really see a way to do it cleanly without refactoring that proc, by the looks of it

Reply:
Procs refactored